cmd/proxy-server/server/server.go

```diff
@@ -297,3 +312,37 @@ func Run(ctx context.Context, opts ...Option) {
 		os.Exit(1)
 	}
 }
+
+// extracts transport credentials from runState. Will error if both credSource and tlsConfig are both specified
+func extractClientTransportCredentialsFromRunState(ctx context.Context, rs *runState) (credentials.TransportCredentials, error) {
+	var creds credentials.TransportCredentials
+	var err error
+	if rs.credSource != "" && rs.tlsConfig != nil {
+		return nil, fmt.Errorf("both credSource and tlsConfig are defined for the client")
+	} else if rs.credSource != "" {
```

Comment:
Drop the else and outdent this if as the above case is terminal if it's taken (no need for the else then)